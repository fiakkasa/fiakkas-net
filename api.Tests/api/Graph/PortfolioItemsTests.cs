using HotChocolate.Execution;

namespace api.Tests;

public class PortfolioItemsTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task PortfolioItems_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  portfolioItems(order: { createdAt: ASC }, take: 1, where: { version: { eq: 1 } }) {
    totalCount
    items {
      createdAt
      href
      id
      ordinal
      technologiesSummary
      title
      updatedAt
      version
      year
      category {
        createdAt
        href
        id
        title
        updatedAt
        version
      }
      customer {
        createdAt
        href
        id
        title
        updatedAt
        version
      }
      technologies(order: { createdAt: ASC }, take: 1, where: { version: { eq: 1 } }) {
        totalCount
        items {
          createdAt
          href
          id
          title
          updatedAt
          version
        }
        pageInfo {
          hasNextPage
          hasPreviousPage
        }
      }
    }
    pageInfo {
      hasNextPage
      hasPreviousPage
    }
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }
}
