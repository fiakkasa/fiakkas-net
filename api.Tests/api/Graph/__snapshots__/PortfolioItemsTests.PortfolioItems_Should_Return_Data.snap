{
  "data": {
    "portfolioItems": {
      "totalCount": 1,
      "items": [
        {
          "createdAt": "2024-01-01T00:00:00.000Z",
          "href": "/test",
          "id": "28e483e4-6961-4b25-88a9-d1d0a5161109",
          "ordinal": 1,
          "technologiesSummary": "Title",
          "title": "Title",
          "updatedAt": null,
          "version": 1,
          "year": 2024,
          "category": {
            "createdAt": "2024-01-01T00:00:00.000Z",
            "id": "38e483e4-6961-4b25-88a9-d1d0a5161109",
            "title": "Title",
            "updatedAt": null,
            "version": 1
          },
          "customer": {
            "createdAt": "2024-01-01T00:00:00.000Z",
            "href": "/test",
            "id": "18e483e4-6961-4b25-88a9-d1d0a5161109",
            "title": "Title",
            "updatedAt": null,
            "version": 1
          },
          "technologies": {
            "totalCount": 1,
            "items": [
              {
                "createdAt": "2024-01-01T00:00:00.000Z",
                "href": "/test",
                "id": "48e483e4-6961-4b25-88a9-d1d0a5161109",
                "title": "Title",
                "updatedAt": null,
                "version": 1
              }
            ],
            "pageInfo": {
              "hasNextPage": false,
              "hasPreviousPage": false
            }
          }
        }
      ],
      "pageInfo": {
        "hasNextPage": false,
        "hasPreviousPage": false
      }
    }
  }
}
